src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -395,14 +404,18 @@ internal static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, Ra
             }
             else
             {
-                deltaStart = currentRange.Range.Start.Character;
+                deltaStart = currentRange.Range.StartCharacter;
             }
 
             data.Add(deltaLine);
             data.Add(deltaStart);
 
             // length
-            var textSpan = currentRange.Range.AsTextSpan(sourceText);
+            var textSpan = new Range()
+            {
+                End = new Position(currentRange.Range.EndLine, currentRange.Range.EndCharacter),
+                Start = new Position(currentRange.Range.StartLine, currentRange.Range.StartCharacter),
+            }.AsTextSpan(sourceText);
```

Comment:
`AsTextSpan` here needed `Range`

- [ ] TODO: improve

Reply:
I would look at the `AsTextSpan` method and see what its doing here. It looks like all that is needed is to get the `Length`, so it might be simpler to just use the methods on `sourceText` to get the absolute position and do a quick subtraction.